addons/account_edi/wizard/account_invoice_send.py

```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+import base64
+
+
+class AccountInvoiceSend(models.TransientModel):
+    _inherit = 'account.invoice.send'
+    _description = 'Account Invoice Send'
+
+    # TODO make edi_format_ids editable and recompute EDI (attachment or pdf embedding)
+    edi_format_ids = fields.Many2many(related='invoice_ids.journal_id.edi_format_ids', string="Electronic invoicing")
+    edi_attachment_ids = fields.Many2many('ir.attachment', compute='_compute_edi_attachment_ids')
+
+    @api.depends('edi_format_ids')
+    def _compute_edi_attachment_ids(self):
+        for r in self:
+            existing_attachments = self.env['ir.attachment'].search([
+                ('res_model', '=', 'account.move'),
+                ('res_id', 'in', r.invoice_ids.ids),
+                ('edi_format_id', '!=', False)])
+            r.edi_attachment_ids = existing_attachments
+
+            requested_formats = r.edi_format_ids
+            for invoice in r.invoice_ids:
+                available_formats = existing_attachments.filtered(lambda a: a.res_id == invoice._origin.id).edi_format_id
+                missing_formats = requested_formats.filtered(lambda f: f._origin.id not in available_formats.ids)
```

Comment:
I'm not sure, strictly speaking, but if I don't do that, it doesn't work.

- For available_formats : r.invoice_ids.ids is a list of <NewId> so the comparison with res_id always fails.
- For missing_formats : requested formats is also a list of <NewId> whereas available_formats is ids (int). Probably because r.edi_format_ids is a many2many in a TransientModel so I'm guessing the m2m is only in cache.